### DISCUSSION TITLE
realtime: update when field changed

### DISCUSSION THREAD

**Discussion by maxNodeJs** [pocketbase/pocketbase#2620] [Q&A] [answered]
I need some help to resolve issue with realtime/subscription.

I am building system that many companies can use, and all the tables have API roles as set below.

Tenants collection have one role only @request.auth.id != "" and is generated when a user signup as a company.

All other collections roles - @request.auth.id != "" && @request.auth.tenant = tenant

Apart from Leads collection which has additional field named "partner_tenant" - rules set for this collection in list/search & view as:

@request.auth.id != "" && (@request.auth.tenant = tenant || @request.auth.tenant = partner_tenant) 

@request.auth.tenant = tenant - allows for example tenant1 users to see all their leads.

|| @request.auth.tenant = partner_tenant allows tenant2 users to see all shared leads based on partner_tenant if set by tenant1, otherwise empty and and only tenent1 can see.

Hope this is clear so far.

When setting partner_tenant, the subscription triggers, calling reloadLoadedPages as listed below - tenant2 users can see the shared leads.

However, when tenant1 removes partner_tenant ( cancels tenant2 ) the subscription doesn't triger? - (no action is called) and the lead is still in tenant2 leads until they manually refresh, or creating their own lead which then calls the reloadLoadedPages.

The subscription i use is as follows.

unsubscribeCollection = await ApiClient.collection(collection.id).subscribe('*', async ({ action, record }) => {
    
    console.log(action); // no action is called if the partner_tenant id is deleted by tenant1
    if (action === "update") {
        reloadLoadedPages();
    } .....

may not be relevant however -     
onDestroy() i call unsubscribeCollection?.();

Would appreciate your help

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the description and the provided code sample.

To what collection do you subscribe? Does `collection.id` in the above refers to the leads or tenants collection?

By saying _"when tenant1 removes partner_tenant"_ do you mean when a leads collection record is updated?

Could you export your collections configuration as JSON (_Admin UI > Settings > Export collections_)?

**Reply by maxNodeJs** [reply]
when users on leads view  the collection is set, and therefore collection.id would be leads.
if they change to other collections, such as users - the collection is updated based on the view they choose. same as in pocketbase UI.

In this case is the leads collection - and after update by tenant1 the realtime leads on both tenant1 and tenant2 is triggered , however after removing partner_tenant by tenant1 and update,  the realtime updates triggers on tenant1 leads, but not tenant2 leads.

you: By saying "when tenant1 removes partner_tenant" do you mean when a leads collection record is updated? YES


[pb_schema.zip](https://github.com/pocketbase/pocketbase/files/11626220/pb_schema.zip)

**Reply by ganigeorgiev** [reply, marked as answer]
I think your use case is similar to https://github.com/pocketbase/pocketbase/issues/2522.

Since you are subscribing to the `*` topic, if the request client no longer satisfy the collection List API rule, then they'll not receive an event about the change.
